modules/apps/27-interchain-accounts/types/codec.go

```diff
@@ -25,7 +26,7 @@ func RegisterInterfaces(registry codectypes.InterfaceRegistry) {
 // SerializeCosmosTx serializes a slice of sdk.Msg's using the CosmosTx type. The sdk.Msg's are
 // packed into Any's and inserted into the Messages field of a CosmosTx. The proto marshaled CosmosTx
 // bytes are returned. Only the ProtoCodec is supported for serializing messages.
-func SerializeCosmosTx(cdc codec.BinaryCodec, msgs []sdk.Msg) (bz []byte, err error) {
+func SerializeCosmosTx(cdc codec.BinaryCodec, msgs []proto.Message) (bz []byte, err error) {
```

Comment:
So this allows the inclusion of a go modules dependency or manually copied proto definition into a chain binary, but does not require the msg types in that dependency to be registered against `sdk.Msg` in the `InterfaceRegistery` - is that correct?

If a non `sdk.Msg` implementation is passed here it will fail on the host on deserialization.

Reply:
Correct! The test case I added receives this error: `no concrete type registered for type URL /cosmos.bank.v1beta1.MsgSendResponse against interface *types.Msg`